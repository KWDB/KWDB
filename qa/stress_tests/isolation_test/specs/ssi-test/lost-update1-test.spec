# test lost update
setup
{
  CREATE TABLE IF NOT EXISTS foo (
	key		int PRIMARY KEY,
	value	int
  );

  INSERT INTO foo VALUES (1000, 1000);
}

teardown
{
  DROP TABLE foo;
}

session s1
step s1		{ BEGIN transaction isolation level SERIALIZABLE; }
step s1r	{ SELECT * FROM foo; }
step s1u  { UPDATE foo SET key=key-200; }
step s1c	{ COMMIT; }

session s2
step s2		{ BEGIN transaction isolation level SERIALIZABLE; }
step s2r  { SELECT * FROM foo; }
step s2u	{ UPDATE foo SET key=key+100; }
step s2c	{ ROLLBACK; }

permutation s1 s2 s1u s2u s1c s2c s1r s2r


