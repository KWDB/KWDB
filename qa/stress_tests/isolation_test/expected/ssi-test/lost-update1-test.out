Parsed test spec with 2 sessions

starting permutation: s1 s2 s1u s2u s1c s2c s1r s2r
step s1: BEGIN transaction isolation level SERIALIZABLE;  finished.
step s2: BEGIN transaction isolation level SERIALIZABLE;  finished.
step s1u: UPDATE foo SET key=key-200;  finished.
step s2u: UPDATE foo SET key=key+100; <waiting ...>
step s1c: COMMIT;  finished.
step s2u: <... completed>
step s2c: ROLLBACK;  finished.
step s1r: SELECT * FROM foo;  finished.
key|value
---+-----
800| 1000
(1 row)

step s2r: SELECT * FROM foo;  finished.
key|value
---+-----
800| 1000
(1 row)

